internal/handlers/pg/msg_find.go

```diff
@@ -154,3 +169,65 @@ func (h *Handler) MsgFind(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, er
 
 	return &reply, nil
 }
+
+func getMaxTimeMSParameter(document *types.Document) (int32, error) {
+	var maxTimeMS int32
+	maxTimeMSParam, err := common.GetOptionalParam(document, "maxTimeMS", maxTimeMS)
+	if err != nil {
+		var commonErr *common.Error
+		if errors.As(err, &commonErr) {
+			return 0, getErrorForInvalidMaxTimeMS(document)
+		}
+
+		return 0, err
+	}
+
+	if maxTimeMSParam < 0 {
+		return 0, common.NewErrorMsg(
+			common.ErrBadValue,
+			fmt.Sprintf("%v value for maxTimeMS is out of range", maxTimeMSParam),
+		)
+	}
+
+	return maxTimeMSParam, nil
+}
+
+func getErrorForInvalidMaxTimeMS(document *types.Document) error {
+	v, err := document.Get("maxTimeMS")
```

Comment:
Why the code above used `common.GetOptionalParam(document, "maxTimeMS", maxTimeMS)` to get the value and here it's `document.Get`? What's the difference?

Reply:
Because we need to check the type and return the corresponding value in the first case it is converted to type int32 and we need to check the type, so we need to get this value again and already determine the type